src/modules/ModuleRunner.ts

```diff
@@ -66,13 +82,67 @@ export class ModuleRunner {
         return merged;
     }
 
+    /**
+     * Ensure we register extensions provided by the modules
+     */
+    private updateExtensions(): void {
+        const cryptoSetupExtensions: Array<RuntimeModule> = [];
+        const experimentalExtensions: Array<RuntimeModule> = [];
+
+        this.modules.forEach((m) => {
+            /* Record the cryptoSetup extensions if any */
+            if (m.module.extensions?.cryptoSetup) {
+                cryptoSetupExtensions.push(m.module);
+            }
+
+            /* Record the experimantal extensions if any */
+            if (m.module.extensions?.experimental) {
+                experimentalExtensions.push(m.module);
+            }
+        });
+
+        /* Enforce rule that only a single module may provide a given extension */
+        if (cryptoSetupExtensions.length > 1) {
+            throw new Error(
+                `cryptoSetup extension is provided by modules ${cryptoSetupExtensions
+                    .map((m) => m.moduleName)
+                    .join(", ")}, but can only be provided by a single module`,
+            );
+        }
+        if (experimentalExtensions.length > 1) {
+            throw new Error(
+                `experimental extension is provided by modules ${experimentalExtensions
+                    .map((m) => m.moduleName)
+                    .join(", ")}, but can only be provided by a single module`,
+            );
+        }
+
+        /* Override the default extension if extension was provided by a module */
+        if (cryptoSetupExtensions.length == 1) {
+            this.extensions.cryptoSetup = cryptoSetupExtensions[0].extensions?.cryptoSetup;
+        }
+
+        if (experimentalExtensions.length == 1) {
+            this.extensions.experimental = cryptoSetupExtensions[0].extensions?.experimental;
+        }
+    }
+
     /**
      * Registers a factory which creates a module for later loading. The factory
      * will be called immediately.
      * @param factory The module factory.
      */
     public registerModule(factory: ModuleFactory): void {
         this.modules.push(new AppModule(factory));
+
+        /**
+         * Check if the new module provides any extensions, and also ensure a given extension is only provided by a single runtime module
+         * Slightly inefficient to do this on each registration, but avoids changes to element-web installer code
+         * Also note that this require that the statement in the comment above, about immediately calling the factory, is in fact true
+         * (otherwise wrapped RuntimeModules will not be available)
+         */
+
```

Comment:
Minor formatting nits:

```suggestion
        /*
         * Check if the new module provides any extensions, and also ensure a given extension is only provided by a single runtime module.
         *
         * Slightly inefficient to do this on each registration, but avoids changes to element-web installer code.
         *
         * Also note that this requires that the statement in the comment above, about immediately calling the factory, is in fact true
         * (otherwise wrapped RuntimeModules will not be available).
         */
```

Reply:
Thanks. 
The code will be refactored, and this comment will change accordingly.